pytorch_lightning/cluster_environments/lsf_environment.py

```diff
@@ -0,0 +1,176 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import socket
+import warnings
+from pytorch_lightning import _logger as log
+from pytorch_lightning.cluster_environments.cluster_environment import ClusterEnvironment
+
+
+class LSFEnvironment(ClusterEnvironment):
```

Comment:
From your knowledge, do you think it would be worth to add ci / cd for the LSF ClusterEnvironment. It seemed possible using https://github.com/IBMSpectrumComputing/lsf-hybrid-cloud/tree/master/LSF_On_AWS.

Reply:
Being able to test this would be nice, but I'm not aware of any CI service that uses LSF. 

That is for extending an existing LSF system to use cloud resources, so I don't think it would work as a CI/CD solution for us.